tests/services/test_record_service.py

```diff
@@ -385,3 +385,32 @@ def test_reindexing_all_siblings(app, service, identity_simple, input_data):
     hits = [hit for hit in service.draft_cls.index.search() if hit["id"] == recid]
     assert hits
     assert hits[0]["metadata"]["title"] == draft.metadata["title"]
+
+
+def test_new_draft_of_published_record_doesnt_override_next_draft_id(
+        app, service, identity_simple, input_data):
+
+    # These imports will be placed at top
+    # this is just until we get something working
+    from mock_module.api import Draft
+    # create published record
+    record = create_and_publish(service, identity_simple, input_data)
+    recid = record.id
+
+    Draft.index.refresh()
+
+    # cleanup_drafts
+    from datetime import timedelta
+    service.cleanup_drafts(timedelta(0))
+
+    Draft.index.refresh()
+
+    # create a new version - can comment/uncomment to try out combinations
+    new_version_draft = service.new_version(identity_simple, recid)
+    assert record._record.versions.next_draft_id == new_version_draft._record.id
+    assert record._record.versions.latest_id == record._record.id
+
+    # create a draft of published record
+    draft = service.edit(identity_simple, record.id)
```

Comment:
This is the test to look into that exhibits the following behavior:

opensearchpy.exceptions.ConflictError: ConflictError(409, 'version_conflict_engine_exception', '[95283d9d-e29e-4b5f-a9a2-6d6a7b1dac1c]: version conflict, current version [3] is higher than the one provided [1]')
[4:02 PM]

My understanding: The fresh new draft for the existing record is conflicting in the index with the old draft of the same record. This seems to be because cleanup_drafts() doesn't delete the draft document in the index. But that is already strange because when interacting with a local running instance, this conflict error doesn't show up.
[4:02 PM]
So I added self.draft_indexer.delete(d) for each deleted draft d, but that fails with :

NotFoundError(404, '{"_index":"draftsresources-drafts-draft-v1.0.0-1688067005","_id":"95283d9d-e29e-4b5f-a9a2-6d6a7b1dac1c","_version":4,"result":"not_found","_shards":{"total":2,"successful":1,"failed":0},"_seq_no":2,"_primary_term":1}')
 (edited)
[4:02 PM]

So... can't delete because it doesn't exist, but can't index because it does exist :/